src/screens/HyperSwitch/UserManagement/PermissionHelper.res

```diff
@@ -0,0 +1,110 @@
+  | UsersRead
+  | UsersWrite
+  | UnknownPermission(string)
+
+let mapPermissionTypeToString = permissionType => {
+  switch permissionType {
+  | PaymentRead => "PaymentRead"
+  | PaymentWrite => "PaymentWrite"
+  | RefundRead => "RefundRead"
+  | RefundWrite => "RefundWrite"
+  | ApiKeyRead => "ApiKeyRead"
+  | ApiKeyWrite => "ApiKeyWrite"
+  | MerchantAccountRead => "MerchantAccountRead"
+  | MerchantAccountWrite => "MerchantAccountWrite"
+  | MerchantConnectorAccountRead => "MerchantConnectorAccountRead"
+  | MerchantConnectorAccountWrite => "MerchantConnectorAccountWrite"
+  | ForexRead => "ForexRead"
+  | RoutingRead => "RoutingRead"
+  | RoutingWrite => "RoutingWrite"
+  | DisputeRead => "DisputeRead"
+  | DisputeWrite => "DisputeWrite"
+  | MandateRead => "MandateRead"
+  | MandateWrite => "MandateWrite"
+  | CustomerRead => "CustomerRead"
+  | CustomerWrite => "CustomerWrite"
+  | FileRead => "FileRead"
+  | FileWrite => "FileWrite"
+  | Analytics => "Analytics"
+  | ThreeDsDecisionManagerWrite => "ThreeDsDecisionManagerWrite"
+  | ThreeDsDecisionManagerRead => "ThreeDsDecisionManagerRead"
+  | SurchargeDecisionManagerWrite => "SurchargeDecisionManagerWrite"
+  | SurchargeDecisionManagerRead => "SurchargeDecisionManagerRead"
+  | UsersRead => "UsersRead"
+  | UsersWrite => "UsersWrite"
+  | UnknownPermission(val) => val
+  }
+}
+
+let mapStringToPermissionType = val => {
+  switch val {
+  | "PaymentRead" => PaymentRead
+  | "PaymentWrite" => PaymentWrite
+  | "RefundRead" => RefundRead
+  | "RefundWrite" => RefundWrite
+  | "ApiKeyRead" => ApiKeyRead
+  | "ApiKeyWrite" => ApiKeyWrite
+  | "MerchantAccountRead" => MerchantAccountRead
+  | "MerchantAccountWrite" => MerchantAccountWrite
+  | "MerchantConnectorAccountRead" => MerchantConnectorAccountRead
+  | "MerchantConnectorAccountWrite" => MerchantConnectorAccountWrite
+  | "ForexRead" => ForexRead
+  | "RoutingRead" => RoutingRead
+  | "RoutingWrite" => RoutingWrite
+  | "DisputeRead" => DisputeRead
+  | "DisputeWrite" => DisputeWrite
+  | "MandateRead" => MandateRead
+  | "MandateWrite" => MandateWrite
+  | "CustomerRead" => CustomerRead
+  | "CustomerWrite" => CustomerWrite
+  | "FileRead" => FileRead
+  | "FileWrite" => FileWrite
+  | "Analytics" => Analytics
+  | "ThreeDsDecisionManagerWrite" => ThreeDsDecisionManagerWrite
+  | "ThreeDsDecisionManagerRead" => ThreeDsDecisionManagerRead
+  | "SurchargeDecisionManagerWrite" => SurchargeDecisionManagerWrite
+  | "SurchargeDecisionManagerRead" => SurchargeDecisionManagerRead
+  | "UsersRead" => UsersRead
+  | "UsersWrite" => UsersWrite
+  | val => UnknownPermission(val)
+  }
+}
+
+let getAccessValue = (~permissionValue: permissionType, permissionList) => {
+  open AuthTypes
+  let isPermissionFound =
+    permissionList
+    ->Array.find(ele => {
+      ele === permissionValue
+    })
+    ->Option.getWithDefault(UnknownPermission(""))
+
+  isPermissionFound->mapPermissionTypeToString->String.length > 0 ? Read : NoAccess
```

Comment:
Can we use switch here than checking the string length ?

Reply:
Updated